app/services/analytics.rb

```diff
@@ -7,9 +7,11 @@ def initialize(user, request)
   def track_event(event, attributes = { user_id: uuid })
     attributes[:user_id] = uuid unless attributes.key?(:user_id)
 
-    Rails.logger.info("#{event}: #{attributes}")
+    consolidated_attributes = attributes.merge!(request_attributes)
```

Comment:
just out of curiosity, why the `merge!` -- it's unepected to me to mutate an input